marlowe-playground-client/src/HaskellEditor/State.purs

```diff
@@ -87,8 +88,13 @@ handleAction Compile = do
           let
             mContract :: Maybe Contract
             mContract = (fromTerm <=< hush <<< parseContract) interpretedResult.result
+
+            metadataHints :: MetadataHintInfo
+            metadataHints = maybe mempty getMetadataHintInfo mContract
           in
-            for_ mContract $ (modifying (_analysisState <<< _templateContent)) <<< updateTemplateContent <<< getPlaceholderIds
+            for_ mContract \contract -> do
+              modifying (_analysisState <<< _templateContent) $ updateTemplateContent $ getPlaceholderIds contract
+              assign _metadataHintInfo metadataHints
```

Comment:
This causes 2 renders instead of one. We should be able to use `modify_` + `over` instead of  `modifying` and compose it with `set` instead of assign to do the changes in a single pass.

Reply:
I didn't check if it compiles, but something on the lines of:

```purescript
for_ mContract \contract -> do
    modify_ 
      $ over (_analysisState <<< _templateContent) (updateTemplateContent $ getPlaceholderIds contract)
        <<< set _metadataHintInfo metadataHints
```

Reply:
Don't know how to use `modify_` but I have combined them using `modifying` which should do the trick:
```haskell
modifying identity
  $ over (_analysisState <<< _templateContent) (updateTemplateContent $ getPlaceholderIds contract)
  <<< set _metadataHintInfo metadataHints
```